src/fairseq2/recipes/lm/preference_units/preference_criterion_config.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union
+
+from fairseq2.typing import DataType
+
+
+@dataclass
+class PreferenceCriterionConfig:  # TODO: should this be abstract?
+    """Holds the variant-configurations of a language model preference-finetuning task."""
+
+    # Reference Model
+    reference_model: Union[str, Path]
```

Comment:
There are preference optimization techniques that do not use any reference model. My suggestion is to just delete this class since there is no attribute that is common to all techniques.